### PR TITLE
Vault Overhaul

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Vault was poorly designed and an antag deathtrap. This PR changes it and the contents inside so it's desirable for some antags.

https://i.imgur.com/XizcN0J.png

The vault has random loot as well. It chooses 3 random rigsuits from a list, 3 random rigsuit modules from a list, 3 random hostile AI modules from a list, 4 random money-related loot from a list (gold, diamonds, cash, ect), 3 random telecrystal stacks, (up to 75 total; very rare), 3 cardox grenades, and one hand teleporter.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
Reason why the main level has so many commits is that the key error happened, again, and I had to fix that, again.

**Comment by BurgerLUA** [comment]
!review

**Comment by Sindorman** [comment]
hmmm I would approve these games. But do you have roof on top turfs that do not have walls, wouldn't antags be able to get into between walls from level above or is there a roof for it?

**Comment by BurgerLUA** [comment]
@Sindorman There is a way for the antags to get through the top, yes. I'm not entirely sure if I should allow it, but I'm going to leave it just to see.

If antags end up exploiting one method of getting in then I will be patching it, likely, because I'm that kind of asshole.

**Comment by Sindorman** [comment]
Alright, I would say we can test it.

File: code/game/objects/random/random.dm [hunk 1]
```diff
@@ -1091,3 +1094,89 @@
...
+	spawnlist = list(
+		/obj/item/weapon/aiModule/antimov = 1,
+		/obj/item/weapon/aiModule/asimov = 1,
+		/obj/item/weapon/aiModule/purge = 1,
+		/obj/item/weapon/aiModule/quarantine = 1,
+		/obj/item/weapon/aiModule/freeform = 2,
+		/obj/item/weapon/aiModule/freeformcore = 1,
+		/obj/item/weapon/aiModule/oneHuman = 0.5,
+		/obj/item/weapon/aiModule/oxygen = 1,
+		/obj/item/weapon/aiModule/syndicate = 1
```
Comment (Arrow768) [inline code/game/objects/random/random.dm]: NT wouldnt have a syndicate module in the vault

File: code/game/objects/random/random.dm [hunk 2]
```diff
@@ -1091,3 +1094,89 @@
...
+	desc = "Contains a random evil AI module."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "std_mod"
+	spawnlist = list(
+		/obj/item/weapon/aiModule/antimov = 1,
+		/obj/item/weapon/aiModule/asimov = 1,
+		/obj/item/weapon/aiModule/purge = 1,
+		/obj/item/weapon/aiModule/quarantine = 1,
+		/obj/item/weapon/aiModule/freeform = 2,
+		/obj/item/weapon/aiModule/freeformcore = 1,
```
Comment (Arrow768) [inline code/game/objects/random/random.dm]: For us the freeform module is effectively the same as the freeformcore module.
Get rid of the freeformcore module and reduce its likelyhood

File: code/game/objects/random/random.dm (hunk 1 above)
Comment (BurgerLUA) [reply code/game/objects/random/random.dm]: Done.

File: code/game/objects/random/random.dm (hunk 2 above)
Comment (BurgerLUA) [reply code/game/objects/random/random.dm]: Done.

**Comment by skull132** [comment]
\> Feature PR taggged with "bugfix"

but why tho

**Comment by Sindorman** [comment]
Technically fixes #5620 @skull132

**Pull request merged**